pennylane_qiskit/qiskit_device.py

```diff
@@ -283,8 +300,16 @@ def apply_operations(self, operations):
 
         for operation in operations:
             # Apply the circuit operations
-            device_wires = self.map_wires(operation.wires)
-            par = operation.parameters
+            if isinstance(operation, Controlled):
+                device_wires = self.map_wires(operation.base.wires + operation.control_wires)
+            else:
+                device_wires = self.map_wires(operation.wires)
+
+            if isinstance(operation, Barrier):
+                op_num_wires = operation.num_wires
+                par = [self.num_wires] if op_num_wires == -1 else [op_num_wires]
```

Comment:
does anything else in PennyLane ever have `num_wires = -1`? If not (or if it should be treated this way as well), I'd replace all this with:
```python
par = [operation.num_wires]
if par == [-1]:
    par = [self.num_wires]
```
to avoid the `isinstance` call. Then again, maybe comparing arrays is slower... something like that :p

Reply:
The problem here is that the parameters of Qiskit's barrier operation corresponds to the `num_wires` of PL's Barrier operation

Reply:
I suppose that would have been too much generalization without a reason. roger that